pkg/apis/core/validation/shoot.go

```diff
@@ -1046,12 +1049,33 @@ func ValidateWorkers(workers []core.Worker, fldPath *field.Path) field.ErrorList
 			allErrs = append(allErrs, field.Duplicate(fldPath.Index(i).Child("name"), worker.Name))
 		}
 		workerNames[worker.Name] = true
+
+		switch {
+		case atLeastOnePoolWithCompatibleTaints:
+		case len(worker.Taints) == 0:
```

Comment:
```suggestion
		case atLeastOnePoolWithCompatibleTaints, len(worker.Taints) == 0:
```
This would be the correct way to mention multiple cases simultaneously.

Reply:
This was not my intention, although it doesn't make a big difference:
1. If `atLeastOnePoolWithCompatibleTaints` is `true` then nothing is done (first case)
1. If `len(worker.Taints) == 0` is `true` then `atLeastOnePoolWithCompatibleTaints` is set to `true` (second case)
1. Otherwise, the loop is executed (third case).

Reply:
ah yes. Thanks for correcting.